src/components/MultiHopTrade/components/MultiHopTradeConfirm/MultiHopTradeConfirm.tsx

```diff
@@ -26,23 +26,6 @@ export const MultiHopTradeConfirm = memo(() => {
 
   const { isApprovalInitiallyNeeded, isLoading } = useIsApprovalInitiallyNeeded()
 
-  // NOTE: we use dom manipulation to move the component around without losing the component
-  // internal state. If we decide this approach isn't suitable we'll need to rewrite the trade
-  // execution hooks and components to store state and metadata inside redux
-  const containerRef = useRef<HTMLDivElement>(null)
-  const tradeCompleteSummaryRef = useRef<HTMLDivElement>(null)
-  const tradeExecutionRef = useRef<HTMLDivElement>(null)
-  useEffect(() => {
-    const currentTargetRef =
-      tradeExecutionState === MultiHopExecutionState.TradeComplete
-        ? tradeCompleteSummaryRef
-        : tradeExecutionRef
-
-    if (containerRef.current && currentTargetRef.current) {
-      currentTargetRef.current.appendChild(containerRef.current)
-    }
-  }, [tradeExecutionState]) // Rerun when tradeExecutionState changes
```

Comment:
this garbage was the root cause of the crash. Fixed by implementing state management properly (i.e by using redux as a state machine rather than the UI)